exporter/opentelemetry-exporter-datadog/src/opentelemetry/exporter/datadog/exporter.py

```diff
@@ -282,3 +299,27 @@ def _parse_tags_str(tags_str):
             parsed_tags[key] = value
 
     return parsed_tags
+
+
+def _extract_tags_from_resource(resource):
+    """Parse tags from resource.labels, except service.name which
+    has special significance within datadog"""
+    tags = {}
+    service_name = None
+    if not resource and resource.labels:
+        return [tags, service_name]
+
+    for attribute_key, attribute_value in resource.labels.items():
+        if isinstance(attribute_value, (int, bool, float)):
+            value = str(attribute_value)
+        elif isinstance(attribute_value, str):
+            value = attribute_value[:128]
```

Comment:
Had borrowed from the Zipkin exporter implementation of exporting Resource info, but thinking this through, This can probably just not get truncated, i believe the datadog-agent has a 5000 character limit on tag length regadless